File: Angular Application/ProStaff/src/app/services/salary.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { Observable } from 'rxjs';
export interface SalaryLog {
  id: number;
  employeeEmail: string;
  date: string;  // ISO string
  time: string;  // ISO string
  message: string;
}
@Injectable({
  providedIn: 'root'
})
export class SalaryService {

  private baseUrl = 'http://localhost:5555';

  constructor(private http: HttpClient, private router: Router) { }

  getEmployeeSalary(employeeEmail: string): Observable<number> {
    const body = { employeeEmail };
    return this.http.post<number>(`${this.baseUrl}/salary/get-employee-salary`, body);
  }

  addFunds(amount: number): Observable<{ orderId: string }> {
    const url = `${this.baseUrl}/salary/add-funds/${amount}`;
    return this.http.post<{ orderId: string }>(url, {});
  }

  getTotalRemuneration(adminEmail: string, employeeEmails: string[]): Observable<number> {
    const url = `${this.baseUrl}/salary/get-total-remuneration`;  // Update with full URL if needed (e.g., `${environment.apiBaseUrl}/salary/get-total-remuneration`)

    const formData = new FormData();
    const adminEmailWrapper = { "adminEmail": adminEmail }
    formData.append('adminEmailWrapper', new Blob([JSON.stringify(adminEmailWrapper)], { type: 'application/json' }));
    formData.append('employeeEmails', new Blob([JSON.stringify(employeeEmails)], { type: 'application/json' }));


    return this.http.post<number>(url, formData);
  }

  getSalaryLogs(adminEmail: string) {
    const body = { adminEmail };
    return this.http.post<SalaryLog[]>(`${this.baseUrl}/salary/get-salary-logs`, body);
  }

  rolloutSalary(adminEmail: string, employeeEmails: string[]): Observable<boolean> {
    const formData = new FormData();
    const adminEmailWrapper = { "adminEmail": adminEmail }
    formData.append('adminEmailWrapper', new Blob([JSON.stringify(adminEmailWrapper)], { type: 'application/json' }));
    formData.append('employeeEmails', new Blob([JSON.stringify(employeeEmails)], { type: 'application/json' }));
    return this.http.post<boolean>(`${this.baseUrl}/salary/rollout-salary`, formData);
  }


  verifyFundPayment(payload: {
    razorpay_order_id: string;
    razorpay_payment_id: string;
    razorpay_signature: string;
    adminEmail: string;
    amount: number;
  }): Observable<boolean> {
    const url = `${this.baseUrl}/salary/verify-fund-payment`;
    return this.http.post<boolean>(url, payload);
  }


  updateEmployeeSalary(newSalary: number, employeeEmail: string): Observable<boolean> {
    const url = `${this.baseUrl}/salary/update-employee-salary/${newSalary}`;
    const body = {
      employeeEmail: employeeEmail
    };

    return this.http.put<boolean>(url, body);
  }

  getCurrentOrganizationFund(adminEmail: string): Observable<number> {
    const url = `${this.baseUrl}/salary/get-current-organization-fund`;
    const body = {
      adminEmail: adminEmail
    };

    return this.http.post<number>(url, body);
  }
}

File: Spring Boot Microservices/Service-Salary/src/main/java/com/prostaff/service_salary/service/SalaryService.java
package com.prostaff.service_salary.service;

import java.util.List;

import com.prostaff.service_salary.dto.AdminEmailWrapper;
import com.prostaff.service_salary.dto.EmployeeEmailWrapper;
import com.prostaff.service_salary.dto.FundOrderDetails;
import com.prostaff.service_salary.dto.FundsPaymentDetails;
import com.prostaff.service_salary.entity.SalaryLog;

public interface SalaryService {

	// write this comment in service interface “razor pay create order and verify payment methods left” 
	
	public Long getEmployeeSalary(EmployeeEmailWrapper employeeEmailWrapper);
	public Boolean createOrganizationFund(String organizationName);
	public Boolean addEmployeeSalary(EmployeeEmailWrapper employeeEmailWrapper, Long salary);
	public Boolean deleteEmployeeSalary(EmployeeEmailWrapper employeeEmailWrapper);
	public Boolean updateEmployeeSalary(Long newSalary, EmployeeEmailWrapper employeeEmailWrapper);
	public Long getCurrentOrganizationFund(AdminEmailWrapper adminEmailWrapper);
	public Boolean rollOutSalary(AdminEmailWrapper adminEmailWrapper,List<String>employeeEmails);
	public Long getTotalRemuneration(AdminEmailWrapper adminEmailWrapper, List<String>employeeEmails);
	public List<SalaryLog> getSalaryLogs(AdminEmailWrapper adminEmailWrapper);
	public FundOrderDetails addFunds(Long amount);
	public Boolean verifyPayment(FundsPaymentDetails details); 
}
